qdata/parser.py

```diff
@@ -311,57 +425,106 @@ def mixedlist(self, *args):
         return ParsedList(Lists.MIXEDLIST, wires)
 
     def argument(self, *args):
-        # <id> or
-        # <id>[<nninteger>]
+        """An argument is either an id, or an id with a non-negative integer index."""
         return unpack(args)
 
     def explist(self, *args):
+        """Either a single expression, or a comma separated list of expressions"""
         flat_list = flatten(*args)
         return ParsedList(Lists.EXPLIST, flat_list)
 
     def exp(self, *args):
+        """A mathematica expression.
+
+        This includes:
+
+        * A real value
+        * A non-negative integer
+        * Mathematical constants (currently just pi)
+        * A symbolic parameter (simply id, so must always start with a lowercase character)
```

Comment:
is the symbolic parameter something that's now being added to the grammar?

Reply:
Not sure, it's present here:

![image](https://user-images.githubusercontent.com/2959003/97471727-fbbfe700-1983-11eb-84e6-457a8811c559.png)

Is this something we added? Or was it from the original spec?

Reply:
ahh, that's just a synonym for the function that should be invoked

Reply:
Oh, that's what confused me!